presto-main/src/test/java/io/prestosql/sql/query/TestLateral.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.query;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+public class TestLateral
+{
+    private QueryAssertions assertions;
+
+    @BeforeClass
+    public void init()
+    {
+        assertions = new QueryAssertions();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        assertions.close();
+        assertions = null;
+    }
+
+    @Test
+    public void testUncorrelatedLateral()
+    {
+        assertions.assertQuery(
+                "SELECT * FROM LATERAL (VALUES 1, 2, 3)",
+                "VALUES 1, 2, 3");
+
+        assertions.assertQuery(
+                "SELECT * FROM LATERAL (VALUES 1), (VALUES 'a')",
+                "VALUES (1, 'a')");
+
+        assertions.assertQuery(
+                "SELECT * FROM LATERAL (VALUES 1) CROSS JOIN (VALUES 'a')",
+                "VALUES (1, 'a')");
+    }
```

Comment:
You could use this example failing query to show that not every LATERAL on right of join means "correlated":
`SELECT * FROM (VALUES 1) t(a), (SELECT * FROM LATERAL (SELECT a))`

And a test with LATERAL in the context of aliased relation
`SELECT * FROM LATERAL (VALUES 1) t(a)`

Reply:
> not every LATERAL on right of join means "correlated":
> `SELECT * FROM (VALUES 1) t(a), (SELECT * FROM LATERAL (SELECT a))`

That one is technically not on the right side of a join, since it's inside a subquery. In fact, it fails with "`a` cannot be resolved", as the column is not in the scope of the subquery.

Reply:
> That one is technically not on the right side of a join

Yes, I meant this example only for "documentational" reasons. Since we now support LATERAL in different contexts, it should be clear which contexts allow using correlated symbols and which don't. Precisely, that it is only possible if LATERAL _is_ right of join, and not when LATERAL is _contained_ inside right of join.